### PR TITLE
YARA Generator v0.4

### PR DESCRIPTION
**Pull request by Neo23x0**
- processes YAML files to enrich YARA rules and allow for different types of YARA rules (vulnerable driver, malicious driver)
- creates five output files with different conditions, descriptions and scores - see README for details

### CODE REVIEW

**Comment by josehelps** [comment]
hey @Neo23x0 anyway to keep the ability to generate a yara rule per driver? I was using these in a new button we added to the UI for example [here](https://www.loldrivers.io/drivers/47a351ee-8abe-40d8-bc2b-557390fa0945/):

image

I think this is done with the sha256 as the name see: https://github.com/magicsword-io/LOLDrivers/blob/main/bin/yara-generator/yara-generator.py#L157-L160.

**Comment by Neo23x0** [comment]
Yeah, the thing is, you'd need 3 buttons now. 

1. Threat hunting YARA rule
2. Exact YARA rule
3. YARA rule for renamed driver [^1]

[^1]: a renamed driver, e.g. `PROCEXP152.SYS` renamed to `p1.sys` is highly suspicious and often an indicator of malicious activity. However, not every YARA scanner is able to work with the external variables used in these rules. (LOKI and THOR are using these external variables)

**Comment by josehelps** [comment]
@Neo23x0 changed out the logic that renders detections. I thank you for the clarification, what do you think?
image

by default is not expanded:

image

**Review by josehelps — APPROVED** [review]

File: yaml/043773c5-120a-4c6b-8485-8f1f5c47fd3e.yml [hunk 1]
```diff
@@ -8699,3 +8699,4 @@ KnownVulnerableSamples:
         Code Signing CA,1
 Tags:
 - aswArPot.sys
+- avgArPot.sys
```
Comment (Neo23x0) [inline yaml/043773c5-120a-4c6b-8485-8f1f5c47fd3e.yml:8702]: I found this driver in our testing infrastructure being used with a different name

Screenshot 2023-06-14 at 08 29 41

You can also see it on VT that it has been used with multiple names.

https://www.virustotal.com/gui/file/0f016c80c4938fbcd47a47409969b3925f54292eba2ce01a8e45222ce8615eb8/telemetry

Screenshot 2023-06-14 at 08 23 51

I fixed it by using different tags but maybe you want to handle it differently?
E.g. by making the "Filename:" field a list?

Comment (josehelps) [reply yaml/043773c5-120a-4c6b-8485-8f1f5c47fd3e.yml:8702]: @Neo23x0 I think tags for now works well, if this happens very often we can change the filename to an array as you suggested.

**Pull request merged**